sherpa/astro/utils/src/_utils.cc

```diff
@@ -96,88 +97,95 @@ namespace sherpa { namespace astro { namespace utils {
     //
     // The rmf_fold function will validate argument sizes
     //
-    
+
     npy_intp dim = npy_intp( len_counts );
     FloatArrayType counts;
     if ( EXIT_SUCCESS != counts.zeros( 1, &dim ) )
       return NULL;
-    
+
     if ( EXIT_SUCCESS != rmf_fold( source.get_size(), &source[0],
 				   num_groups.get_size(), &num_groups[0],
 				   first_chan.get_size(), &first_chan[0],
 				   num_chans.get_size(), &num_chans[0],
 				   response.get_size(), &response[0],
 				   counts.get_size(), &counts[0],
 				   npy_uintp(offset))) {
-      
+
       PyErr_SetString( PyExc_ValueError,
 		       (char*)"RMF data is invalid or inconsistent" );
       return NULL;
 
     }
 
     return counts.return_new_ref();
-    
+
   }
-      
+
   template <typename FloatArrayType, typename IntArrayType>
   PyObject* do_group( PyObject* self, PyObject* args )
   {
-    
+
     FloatArrayType data;
     FloatArrayType grouped;
     IntArrayType group;
     const char* type;
-    
+
     if ( !PyArg_ParseTuple( args, (char*)"O&O&s",
 			    (converter)convert_to_array< FloatArrayType >,
 			    &data,
 			    (converter)convert_to_array< IntArrayType >,
 			    &group,
 			    &type))
-      
+
     return NULL;
-    
+
     if ( data.get_size() != group.get_size() ) {
       ostringstream err;
       err << "input array sizes do not match, "
 	  << "data: " << data.get_size() << " vs group: " << group.get_size();
       PyErr_SetString( PyExc_TypeError, err.str().c_str() );
       return NULL;
     }
-    
-    if( EXIT_SUCCESS != _do_group(data.get_size(), data,
-				  group.get_size(), group,
-				  grouped, type) ) {
-      PyErr_SetString( PyExc_ValueError,
-		       (char*)"group data is invalid or inconsistent" );
+
+    try {
+      if( EXIT_SUCCESS != _do_group(data.get_size(), data,
+				    group.get_size(), group,
+				    grouped, type) ) {
+	PyErr_SetString( PyExc_ValueError,
+			 (char*)"group data is invalid or inconsistent" );
+	return NULL;
+      }
+    } catch ( std::out_of_range& e ) {
```

Comment:
I don't see the '#include <stdexcept>' header inclusion in this file. Are you pulling it in from elsewhere? If not, it should be added with the other header includes at the top of the file to ensure that the out_of_range exception type is defined

Reply:
I assume - given a frantic google - that this would be

    include <stdexcept>

?

Reply:
yes

Reply:
yes, with a preprocessor macro. So at ~line 25- add:   
#include <stdexcept>